simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/AutoClearedValue.kt

```diff
@@ -0,0 +1,47 @@
+package org.nypl.simplified.ui.catalog
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * https://github.com/android/architecture-components-samples/blob/master/GithubBrowserSample/app/src/main/java/com/android/example/github/util/AutoClearedValue.kt
+ *
+ * A lazy property that gets cleaned up when the fragment's view is destroyed.
+ *
+ * Accessing this variable while the fragment's view is destroyed will throw NPE.
+ */
+class AutoClearedValue<T : Any>(val fragment: Fragment) : ReadWriteProperty<Fragment, T> {
```

Comment:
I think this could be moved into the module `simplified-android-ktx`.

Reply:
Makes sense, I will move it